fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchAllTest.java

```diff
@@ -166,4 +175,114 @@ public static void generateOutput(Resource resource) {
         }
 
     }
+    
+    @Test(groups = { "server-search-all"}, dependsOnMethods = {
+    "testCreatePatient" })
+    public void testSearchAllUsing2TagsAndNoExistingTag() throws Exception {
+        int firstRunNumber;
+        FHIRParameters parameters = new FHIRParameters();
+        // tag88 doesn't exist, this case is created according to a reported test failure.
+        parameters.searchParam("_tag", "http://ibm.com/fhir/tag|tag88,tag2,tag");
+        parameters.searchParam("_count", "1000");
+        parameters.searchParam("_page", "1");
+        FHIRResponse response = client.searchAll(parameters);
+        assertResponse(response.getResponse(), Response.Status.OK.getStatusCode());
+        Bundle bundle = response.getResource(Bundle.class);
+        assertNotNull(bundle);
+
+        firstRunNumber = bundle.getEntry().size();
+        assertTrue(firstRunNumber >= 1);
+        // Create one more patient with 2 tags: "tag" and "tag2".
+        testCreatePatient();
+        response = client.searchAll(parameters);
+        assertResponse(response.getResponse(), Response.Status.OK.getStatusCode());
+        bundle = response.getResource(Bundle.class);
+        // The second run should only have one more record found.
+        // Because we limit the page size to 1000, so we only do this check when firstRunNumber < 1000.
+        if (firstRunNumber < 1000) {
+            assertTrue(bundle.getEntry().size() == firstRunNumber + 1);
+        }
+        
+        //Verify that there is no duplicated resource in the search results.
+        // Just need to do the verification for the second run.
+        HashSet<String> resourceIdSet = new HashSet<String>();  
+        for (Entry entry: bundle.getEntry()) {
+            resourceIdSet.add(entry.getResource().getClass().getSimpleName()
+                    + ":" + entry.getResource().getId().getValue());
+        }
+        assertTrue(bundle.getEntry().size() == resourceIdSet.size());
```

Comment:
did you check if any existing helper methods exist already to cover this?

if not, maybe make this into a helper method?

Reply:
done